### ISSUE TITLE
Does not seem possible to use remove rules

### ISSUE THREAD

**Issue by ftyers** [apertium/apertium-lex-tools#46]
```
^Emplear<vblex><pri><p3><sg>/Emprar<vblex><pri><p3><sg>/Ocupar<vblex><pri><p3><sg>$ ^a<pr>/a<pr>//de<pr>$ ^uno<det><ind><m><sg>/un<det><ind><m><sg>$ ^70%<num>/70%<num>$ ^de<pr>/de<pr>$ ^el<det><def><m><sg>/el<det><def><m><sg>$ ^total<n><m><sg>/total<n><m><sg>$ ^de<pr>/de<pr>$ ^asalariado<n><m><pl>/assalariat<n><m><pl>$^.<sent>/.<sent>$^.<sent>/.<sent>$
```

```
<rules>
  <rule weight="1.0">
    <match lemma="a" tags="pr"><select lemma="a"/></match>
  </rule>
  <rule weight="2.0">
    <match lemma="emplear"/>
    <match lemma="a" tags="pr"><remove tags="pr"/></match>
  </rule>
</rules>
```

```
Recogniser: <remove><ANY_CHAR><pr>, [finals: 1]
Recogniser: <select>a<ANY_TAG>, [finals: 1]
recognisers: 2
outOfWord = false
  step: E
outOfWord = false
  step: m
outOfWord = false
  step: p
outOfWord = false
  step: l
outOfWord = false
  step: e
outOfWord = false
  step: a
outOfWord = false
  step: r
tag <vblex>: -4
outOfWord = false
  step: <ANY_TAG>
tag <pri>: -4
outOfWord = false
  step: <ANY_TAG>
tag <p3>: -4
outOfWord = false
  step: <ANY_TAG>
tag <sg>: -4
outOfWord = false
  step: <ANY_TAG>
trad[0]: Emprar<vblex><pri><p3><sg>
trad[0]: Ocupar<vblex><pri><p3><sg>
[POS] 0: [sl 27 ; tl 2 ; bl 0]: Emplear<vblex><pri><p3><sg>
seen:
#CURRENT_ALIVE: 2
==> new pos: 1
outOfWord = false
  step: a
  step: a
tag <pr>: -7
outOfWord = false
  step: <pr>
  step: <pr>
trad[1]: a<pr>
trad[1]: 
trad[1]: de<pr>
[POS] 1: [sl 5 ; tl 3 ; bl 1]: a<pr>
    filter_finals: /<$><skip><$><remove><ANY_CHAR><pr><$><2>
id:      <2>: (lambda: 2.00000)
op:        <skip>
op:        <remove><ANY_CHAR><pr>
#[1]SCORE 2.00000 / <remove><ANY_CHAR><pr>
    filter_finals: /<$><select>a<ANY_TAG><$><1>
id:      <1>: (lambda: 1.00000)
op:        <select>a<ANY_TAG>
#[1]SCORE 1.00000 / <select>a<ANY_TAG>
seen: <1>  <2> 
#CURRENT_ALIVE: 3
==> new pos: 2
outOfWord = false
  step: u
  step: u
  step: u
outOfWord = false
  step: n
  step: n
  step: n
outOfWord = false
  step: o
  step: o
  step: o
tag <det>: -4
outOfWord = false
  step: <ANY_TAG>
  step: <ANY_TAG>
  step: <ANY_TAG>
tag <ind>: -4
outOfWord = false
  step: <ANY_TAG>
  step: <ANY_TAG>
  step: <ANY_TAG>
tag <m>: -4
outOfWord = false
  step: <ANY_TAG>
  step: <ANY_TAG>
  step: <ANY_TAG>
tag <sg>: -4
outOfWord = false
  step: <ANY_TAG>
  step: <ANY_TAG>
  step: <ANY_TAG>
trad[2]: un<det><ind><m><sg>
[POS] 2: [sl 20 ; tl 1 ; bl 1]: uno<det><ind><m><sg>
seen:
#CURRENT_ALIVE: 1
FLUSH:
✔️  >>> 1 -> <remove><ANY_CHAR><pr> -> 2.00000
✔️  >>> 1 -> <select>a<ANY_TAG> -> 1.00000
❎ >>> 1 -> <remove><ANY_CHAR><pr> -> 2.00000
❎ >>> 1 -> <select>a<ANY_TAG> -> 1.00000
✔️  >>> 1 -> <remove><ANY_CHAR><pr> -> 2.00000
❎ >>> 1 -> <select>a<ANY_TAG> -> 1.00000
1:REMOVE:2.00000:a<pr>:a<pr>
1:REMOVE:2.00000:a<pr>:de<pr>
1:SELECT:1.00000:a<pr>:a<pr>
==> new pos: 1
outOfWord = false
  step: 7
outOfWord = false
  step: 0
outOfWord = false
  step: %
tag <num>: -4
outOfWord = false
  step: <ANY_TAG>
trad[1]: 70%<num>
[POS] 1: [sl 8 ; tl 1 ; bl 1]: 70%<num>
seen:
#CURRENT_ALIVE: 1
FLUSH:
==> new pos: 1
outOfWord = false
  step: d
outOfWord = false
  step: e
tag <pr>: -7
outOfWord = false
  step: <pr>
trad[1]: de<pr>
[POS] 1: [sl 6 ; tl 1 ; bl 1]: de<pr>
seen:
#CURRENT_ALIVE: 1
FLUSH:
==> new pos: 1
outOfWord = false
  step: e
outOfWord = false
  step: l
tag <det>: -4
outOfWord = false
  step: <ANY_TAG>
tag <def>: -4
outOfWord = false
  step: <ANY_TAG>
tag <m>: -4
outOfWord = false
  step: <ANY_TAG>
tag <sg>: -4
outOfWord = false
  step: <ANY_TAG>
trad[1]: el<det><def><m><sg>
[POS] 1: [sl 19 ; tl 1 ; bl 1]: el<det><def><m><sg>
seen:
#CURRENT_ALIVE: 1
FLUSH:
==> new pos: 1
outOfWord = false
  step: t
outOfWord = false
  step: o
outOfWord = false
  step: t
outOfWord = false
  step: a
outOfWord = false
  step: l
tag <n>: -4
outOfWord = false
  step: <ANY_TAG>
tag <m>: -4
outOfWord = false
  step: <ANY_TAG>
tag <sg>: -4
outOfWord = false
  step: <ANY_TAG>
trad[1]: total<n><m><sg>
[POS] 1: [sl 15 ; tl 1 ; bl 1]: total<n><m><sg>
seen:
#CURRENT_ALIVE: 1
FLUSH:
==> new pos: 1
outOfWord = false
  step: d
outOfWord = false
  step: e
tag <pr>: -7
outOfWord = false
  step: <pr>
trad[1]: de<pr>
[POS] 1: [sl 6 ; tl 1 ; bl 1]: de<pr>
seen:
#CURRENT_ALIVE: 1
FLUSH:
==> new pos: 1
outOfWord = false
  step: a
outOfWord = false
  step: s
outOfWord = false
  step: a
outOfWord = false
  step: l
outOfWord = false
  step: a
outOfWord = false
  step: r
outOfWord = false
  step: i
outOfWord = false
  step: a
outOfWord = false
  step: d
outOfWord = false
  step: o
tag <n>: -4
outOfWord = false
  step: <ANY_TAG>
tag <m>: -4
outOfWord = false
  step: <ANY_TAG>
tag <pl>: -4
outOfWord = false
  step: <ANY_TAG>
trad[1]: assalariat<n><m><pl>
[POS] 1: [sl 20 ; tl 1 ; bl 1]: asalariado<n><m><pl>
seen:
#CURRENT_ALIVE: 1
FLUSH:
==> new pos: 1
outOfWord = false
  step: .
tag <sent>: -4
outOfWord = false
  step: <ANY_TAG>
trad[1]: .<sent>
[POS] 1: [sl 7 ; tl 1 ; bl 0]: .<sent>
seen:
#CURRENT_ALIVE: 1
FLUSH:
==> new pos: 1
outOfWord = false
  step: .
tag <sent>: -4
outOfWord = false
  step: <ANY_TAG>
trad[1]: .<sent>
[POS] 1: [sl 7 ; tl 1 ; bl 0]: .<sent>
seen:
#CURRENT_ALIVE: 1
FLUSH:
==> new pos: 1
^Emplear<vblex><pri><p3><sg>/Emprar<vblex><pri><p3><sg>/Ocupar<vblex><pri><p3><sg>$0 ^a<pr>/a<pr>$1 ^uno<det><ind><m><sg>/un<det><ind><m><sg>$2 ^70%<num>/70%<num>$1 ^de<pr>/de<pr>$1 ^el<det><def><m><sg>/el<det><def><m><sg>$1 ^total<n><m><sg>/total<n><m><sg>$1 ^de<pr>/de<pr>$1 ^asalariado<n><m><pl>/assalariat<n><m><pl>$1^.<sent>/.<sent>$1^.<sent>/.<sent>$1
```

**Comment by xavivars** [comment]
Actually, after thinking a bit more about this, I'm not sure this is a good idea (even if it would solve the problems in this case).

The reason why this may not work is that it would potentially leave two superblanks together, that would require "merging".

How would you handle that part?

**Comment by ftyers** [comment]
There is `apertium-posttransfer` now which takes care of superfluous spaces.

**Comment by ftyers** [comment]
This should be fixed in 0f8eb65.
```
$ echo "Emplea a un 70% del total de asalariados." | apertium -d . spa-cat-biltrans | lrx-proc -t -m test.lrx.bin
1:REMOVE:2.00000:a<pr>:3:a<pr>
1:REMOVE:2.00000:a<pr>:2:de<pr>
1:SELECT:1.00000:a<pr>:1:a<pr>
^Emplear<vblex><pri><p3><sg>/Emprar<vblex><pri><p3><sg>/Ocupar<vblex><pri><p3><sg>$ ^a<pr>/$ ^uno<det><ind><m><sg>/un<det><ind><m><sg>$ ^70%<num>/70%<num>$ ^de<pr>/de<pr>$ ^el<det><def><m><sg>/el<det><def><m><sg>$ ^total<n><m><sg>/total<n><m><sg>$ ^de<pr>/de<pr>$ ^asalariado<n><m><pl>/assalariat<n><m><pl>$^.<sent>/.<sent>$^.<sent>/.<sent>$
```

**Comment by ftyers** [comment]
Fixed in 0f8eb65.
